tests/base/src/test/java/org/keycloak/tests/admin/IllegalAdminUpgradeTest.java

```diff
@@ -0,0 +1,800 @@
+                    Assertions.fail("should fail with forbidden exception");
+                } catch (ClientErrorException e) {
+                    assertThat(Response.Status.fromStatusCode(e.getResponse().getStatus()),
+                            is(equalTo(Response.Status.FORBIDDEN)));
+                }
+
+                roles.clear();
+                roles.add(realmImpersonate);
+                try {
+                    realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                    Assertions.fail("should fail with forbidden exception");
+                } catch (ClientErrorException e) {
+                    assertThat(Response.Status.fromStatusCode(e.getResponse().getStatus()),
+                            is(equalTo(Response.Status.FORBIDDEN)));
+                }
+
+                roles.clear();
+                roles.add(realmManageUsers);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).remove(roles);
+
+                roles.clear();
+                roles.add(realmViewUsers);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).remove(roles);
+
+                roles.clear();
+                roles.add(realmQueryUsers);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).remove(roles);
+
+                roles.clear();
+                roles.add(realmQueryGroups);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).remove(roles);
+
+                roles.clear();
+                roles.add(realmQueryClients);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).add(roles);
+                realmClient.realm(REALM_NAME).users().get(realmUser.getId()).roles().clientLevel(client.getId()).remove(roles);
+            }
+        }
+        // test master manageUsers only admin can do with master realm admin roles
+        {
+            ClientRepresentation client = masterClient;
+            try (Keycloak realmClient = adminClientFactory.create()
+                    .realm(MASTER_REALM_NAME).username("userAdmin").password("password").clientId(Constants.ADMIN_CLI_CLIENT_ID).build()) {
```

Comment:
Why was the user for this client changed from `"masterAdmin"` to `"userAdmin"`?

Reply:
@vaceksimon let me check